File: code/template/template.py
import os
import json
import traceback
import pandas as pd
import numpy as np

def userMethod({parameters}, outputWanted = 'Result'):
	_error = ""
	_output = None
	_output = {"output": _output, "error": _error}
	errorCode = ""
	#if the variable errorCode is defined in this code it will appear if an exception is catched
	{Check params}
	try:
		_outputNotChecked = run({parametersInFunction}, outputWanted)
		#_output = checkTypes(_outputNotChecked)
		_output = _outputNotChecked
		
	except Exception as e:
		_output = "Err"
		print(traceback.print_exc())
		_error = "Exception captured. Error: " + str(e) + ", User error code: " + errorCode 
	
	return {"output": _output, "error": _error}


def run({parameters}, outputWanted = None):
	outputs = None
	# Main user code goes here
	# Do not use "_" when defining variables
	#______________ Example __________________
	outputs = []
#{userCode}
	#________________________________
	return [#{outputlist}]

File: code/template/template.m
function output = {taskName}({parameters},outputWanted)
	error = "";
	output = 'None';
	errorCode = "";
	
	if ~exist('outputWanted','var')
		outputWanted =  'None';
	end
	%if the variable errorCode is defined in this code it will appear if an exception is catched
	{Check params}
	try
		values = run({parametersInFunction}, outputWanted);
		output = values;
	catch e
		output = "Err";
		error = "Exception captured. , User error code: " + e.message ;
	end 
	outStruct = struct('output', output, 'error', error);
	output = jsonencode(outStruct);
end


function rawInputs = checkTypes(rawInputs)
	names = fieldnames(rawInputs)
	for i = 1:numel(names)
		input = rawInputs.(names{i})
		if(isa(input,"char"))
			rawInputs.(names{i}) = strcat('"', input, '"')
		elseif(isa(input,"string"))
			rawInputs.(names{i}) = strcat("'", input, "'")
		else
			rawInputs.(names{i}) = input
		end
	end
end

function [{outputsCode}] = run({parameters},outputWanted)
	% Main user code goes here
	% Do not use "_" when defining variables
	%______________ Code __________________
	#{userCode}
	%________________________________
end